### ISSUE TITLE
Interact with built image once the manifest file is created.

### ISSUE THREAD

**Issue by rquadling** [hashicorp/packer#4754]
Hi.

**tl;dr;**

1. Packer builds an image.
2. The manifest post-processor creates the manifest file.
3. I want to tag the code in the image with the AMI ID that's present in the manifest file and push those tags to the repos.

**Long version** 
(I'm wordy, so I wrote the tl;dr; as a humorous attempt to get some suggestions / recommendations / support / laughs / beer).

I have Packer building me an AWS AMI image.

I want to tag the code released within that image with the AMI ID.

I've got the manifest post-processor and I can see the manifest file's content.

I know I can use `jq` to parse the manifest file to extract the AMI ID. But `jq` is not installed in the image that packer is running in (I think I can get that changed - but maybe not).

It took me a while to realise that the when I'm running packer, (using gitlab as the tool that runs packer - set up by a third party, so not really 100% sure on what's what), that the image that is produced is not based upon the local filesystem (or so it seems - if I run `cd / && find . | grep epos`, nothing is found, even though the epos code is in the image. The image is built using Ansible.

I think the shell-local post-processor MAY be of help, but, I can't find any documentation on how to do something with the image AND the manifest.

One option would be to somehow pass the entire manifest file to a script within the built image, but I'm not even sure if the built image is 'running' as such. If so, could I scp the file into the image and then run a command via ssh? If so, that would be a really really useful example to have documented.

But, any suggestions would be greatly appreciated.

**Comment by mwhooker** [comment]
I think you might need to state the problem again, being especially clear about what you want to do in which environment (local machine, source instance, etc).

You should be able to get the AMI locally using the manifest and the shell-local post processor.

You can't run anything inside the source instance that uses the AMI ID because the AMI doesn't exist when the instance is available.

You might also be able to get more help on the mailing list: https://groups.google.com/forum/#!forum/packer-tool

**Comment by SwampDragons** [comment]
Going through old issues; I think what you're asking for isn't possible -- by the time we create an ami id, the instance has been shut down, so you can't add that id to code inside of the instance.
